modules/library/coverage/src/main/java/org/geotools/image/util/ImageUtilities.java

```diff
@@ -1276,11 +1291,12 @@ public static void disposeSinglePlanarImage(PlanarImage planarImage) {
         }
 
         try {
-            if (planarImage instanceof RenderedImageAdapter) {
-                cleanField(planarImage, "theImage");
-            }
+            // Check extended class first
             if (planarImage instanceof WritableRenderedImageAdapter) {
                 cleanField(planarImage, "theWritableImage");
+                cleanField(planarImage, "theImage", true);
+            } else if (planarImage instanceof RenderedImageAdapter) {
```

Comment:
Before the two ifs were not alternative, now they are. I guess this is correct as the two fields in question are pointing to the same image right?

Reply:
right.